servicetalk-traffic-resilience-http/src/main/java/io/servicetalk/apple/traffic/resilience/http/TrafficResilienceHttpServiceFilter.java

```diff
@@ -0,0 +1,688 @@
+                        ctx;
+                return applyCapacityControl(capacityPartitions, circuitBreakerPartitions, actualContext, request,
+                        responseFactory, request1 -> delegate().handle(ctx, request1, responseFactory));
+            }
+        });
+    }
+
+    @Override
+    protected Ticket wrapTicket(@Nullable final ServerListenContext serverListenContext, final Ticket ticket) {
+        return serverListenContext == null ? ticket : new ServerResumptionTicketWrapper(serverListenContext, ticket);
+    }
+
+    @Override
+    protected Single<StreamingHttpResponse> handleLocalCapacityRejection(
+            @Nullable final ServerListenContext serverListenContext,
+            final StreamingHttpRequest request,
+            @Nullable final StreamingHttpResponseFactory responseFactory) {
+        assert serverListenContext != null;
+        if (rejectionPolicy.onLimitStopAcceptingConnections) {
+            serverListenContext.acceptConnections(false);
+        }
+
+        if (responseFactory != null) {
+            return rejectionPolicy.onLimitResponseBuilder
+                    .apply(request, responseFactory)
+                    .map(resp -> {
+                        rejectionPolicy.onLimitRetryAfter.accept(resp);
+                        return resp;
+                    });
+        }
+
+        return DEFAULT_CAPACITY_REJECTION;
+    }
+
+    @Override
+    protected Single<StreamingHttpResponse> handleLocalBreakerRejection(
+            final StreamingHttpRequest request,
+            @Nullable final StreamingHttpResponseFactory responseFactory,
+            @Nullable final CircuitBreaker breaker) {
+        if (responseFactory != null) {
+            return rejectionPolicy.onOpenCircuitResponseBuilder
+                    .apply(request, responseFactory)
+                    .map(resp -> {
+                        rejectionPolicy.onOpenCircuitRetryAfter
+                                .accept(resp, new StateContext(breaker));
+                        return resp;
+                    })
+                    .shareContextOnSubscribe();
+        }
+
+        return DEFAULT_BREAKER_REJECTION;
+    }
+
+    /**
+     * Default response rejection policy.
+     * <ul>
+     *     <li>When a request is rejected due to capacity, the service will respond
+     *     {@link RejectionPolicy#tooManyRequests()}.</li>
+     *     <li>When a request is rejected due to capacity, the service will NOT include a retry-after header.</li>
+     *     <li>When a request is rejected due to breaker, the service will respond
+     *     {@link RejectionPolicy#serviceUnavailable()}.</li>
+     *     <li>When a request is rejected due to breaker, the service will respond with Retry-After header hinting
+     *     the duration the breaker will remain open.</li>
+     * </ul>
+     *
+     * @return The default {@link RejectionPolicy}.
+     */
+    public static RejectionPolicy defaultRejectionResponsePolicy() {
```

Comment:
To me, this method is quite hard to discover. None of the other filters have a static method that can return any state or default. It's an unexpected patter.

Should we consider making `RejectionPolicy` a top level class and either providing a static method there or creating `RejectionPolicies` class as we usually do?

Reply:
Moved, but since its only one policy i kept it static there. We can always add a class later as needed.